### PR TITLE
Create GridMergeApp with GridAppTemplate

### PR DESCRIPTION
**Pull request by med-hedi**
Signed-off-by: Mohamed Hedi Khefacha <mohamed.hedi@rte-france.com>

### CODE REVIEW

**Review by geofjamg — CHANGES_REQUESTED** [review]
You have to update everywhere needed with the new app name and not let it as it is.

File: app-httpd.conf [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    RewriteCond /usr/local/apache2/htdocs/gridapp%{REQUEST_URI} -f [OR]
+    RewriteCond %{REQUEST_URI} ^/api/.* [OR]
+    RewriteCond %{REQUEST_URI} ^/ws/.*
+    RewriteRule ^ - [S=2]
+    RewriteRule ^ "${remapbase:%{THE_REQUEST}}" [DPI]
+    RewriteRule "^(.*)$" /index.html [E=BASE:$1,L]
+    SetEnvIf BASE ^$ BASE=.
+
+    ProxyRequests off
+    ProxyPassMatch "^/api/(case-server|study-server|gateway)/(.*)$" http://$1/$2
```
Comment (med-hedi) [inline app-httpd.conf]: @geofjamg I haven't changed it because I'm not sure there is  **merge-server | getway**

File: src/setupProxy.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+const { createProxyMiddleware } = require('http-proxy-middleware');
+module.exports = function(app) {
+  app.use(
+    createProxyMiddleware('http://localhost:5001/api/study-server', {
```
Comment (med-hedi) [inline src/setupProxy.js]: @geofjamg I haven't changed it because I'm not sure there is **api/merge-server**

**Comment by med-hedi** [comment]
Done

File: app-httpd.conf (hunk 1 above)
Comment (geofjamg) [reply app-httpd.conf]: it is merge-orchestrator-server

File: src/setupProxy.js (hunk 2 above)
Comment (geofjamg) [reply src/setupProxy.js]: it is merge-orchestrator-server

File: .env [hunk 3]
```diff
@@ -0,0 +1,3 @@
+REACT_APP_USE_AUTHENTICATION=true
+
+REACT_APP_NAME=GridApp
```
Comment (geofjamg) [inline .env]: To update

File: .env.development [hunk 4]
```diff
@@ -0,0 +1,3 @@
+REACT_APP_USE_AUTHENTICATION=false
+
+REACT_APP_NAME=GridApp
```
Comment (geofjamg) [inline .env.development]: To update

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+          CI: false
+        run: |
+          npm install
+          npm run-script build
+          npm run-script test
+      - name: Build and publish Docker image
+        if: github.ref == 'refs/heads/master'
+        uses: elgohr/Publish-Docker-Github-Action@master
+        with:
+          name: powsybl/gridapp
```
Comment (geofjamg) [inline .github/workflows/build.yml]: To update

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+          username: powsyblci
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Broadcast update event
+        if: github.ref == 'refs/heads/master'
+        uses: gridsuite/broadcast-event@master
+        with:
+          token: ${{ secrets.REPO_ACCESS_TOKEN }}
+          event-type: gridapp_updated
```
Comment (geofjamg) [inline .github/workflows/build.yml]: To update

File: CHANGELOG.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+## [Unreleased]
+
+## [0.1.0] - yyyy-mm-dd
+
+### Added
+
+...
+
+[unreleased]: https://github.com/gridsuite/gridapp-template/compare/v0.2.0...HEAD
```
Comment (geofjamg) [inline CHANGELOG.md]: To update

File: app-httpd.conf [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    DocumentRoot /usr/local/apache2/htdocs/gridapp/
+    AddOutputFilterByType INCLUDES text/html
+    <Directory /usr/local/apache2/htdocs/gridapp/>
+        Options +Includes
+    </Directory>
+
+    RewriteEngine On
+    RewriteMap remapbase "prg:/bin/sed -u -e 's;[^ ]* ;;' -e 's; .*;;' -e 's;[^/]*;..;g' -e 's;../..;;' -e 's/.//' -e 's;^$;.;'"
+
+    RewriteCond /usr/local/apache2/htdocs/gridapp%{REQUEST_URI} -f [OR]
```
Comment (geofjamg) [inline app-httpd.conf]: To update

**Review by geofjamg — CHANGES_REQUESTED** [review]

File: app-httpd.conf [hunk 9]
```diff
@@ -0,0 +1,28 @@
+<VirtualHost *:80>
+    LoadModule proxy_module modules/mod_proxy.so
+    LoadModule proxy_http_module modules/mod_proxy_http.so
+    LoadModule proxy_wstunnel_module modules/mod_proxy_wstunnel.so
+    LoadModule rewrite_module modules/mod_rewrite.so
+    LoadModule include_module modules/mod_include.so
+
+    DocumentRoot /usr/local/apache2/htdocs/gridapp/
```
Comment (geofjamg) [inline app-httpd.conf]: To update

File: app-httpd.conf [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    AddOutputFilterByType INCLUDES text/html
+    <Directory /usr/local/apache2/htdocs/gridapp/>
```
Comment (geofjamg) [inline app-httpd.conf]: To update

**Review by geofjamg — CHANGES_REQUESTED** [review]

File: .env (hunk 3 above)
Comment (med-hedi) [reply .env]: Done

File: .env.development (hunk 4 above)
Comment (med-hedi) [reply .env.development]: Done

File: .github/workflows/build.yml (hunk 5 above)
Comment (med-hedi) [reply .github/workflows/build.yml]: Done

File: .github/workflows/build.yml (hunk 6 above)
Comment (med-hedi) [reply .github/workflows/build.yml]: Done

File: CHANGELOG.md (hunk 7 above)
Comment (med-hedi) [reply CHANGELOG.md]: Done

File: app-httpd.conf (hunk 8 above)
Comment (med-hedi) [reply app-httpd.conf]: Done

File: app-httpd.conf (hunk 9 above)
Comment (med-hedi) [reply app-httpd.conf]: Done

File: app-httpd.conf (hunk 10 above)
Comment (med-hedi) [reply app-httpd.conf]: Done

**Review by geofjamg — APPROVED** [review]

**Pull request merged**